components/form/LifecycleHooks.vue

```diff
@@ -0,0 +1,80 @@
+<script>
+import Vue from 'vue';
+
+import HookOption from '@/components/form/HookOption';
+import { _VIEW } from '@/config/query-params';
+
+export default {
+  components: { HookOption },
+
+  props: {
+    mode: {
+      type:     String,
+      required: true,
+    },
+
+    // container spec
+    value: {
+      type:    Object,
+      default: () => {
+        return {};
+      }
+    },
+  },
+
+  data() {
+    const { postStart, preStop } = this.value;
+
+    return {
+      postStart, preStop, hookOptions: ['postStart', 'preStop']
+    };
+  },
+
+  computed: {
+    isView() {
+      return this.mode === _VIEW;
+    },
+  },
+
+  methods: {
+    update() {
+      const out = {
+        postStart: this.postStart,
+        preStop:   this.preStop
+      };
+
+      for (const prop in out) {
+        const val = out[prop];
+
+        if (val === '' || typeof val === 'undefined' || val === null || val === {}) {
```

Comment:
`val === {}` Wont work as intended. So if we toggle from having a lifecycle method defined to the 'none' radio option, this block of code isn't hit and our container ends up with an empty object for one of the two lifecycle props, which causes an api error. Try using `isEmpty(val)` it's a utility function you can import from  '@/utils/object'

Reply:
I added `isEmpty(val)` to the `update` method, it's now properly removing the two lifecycle props. It will leave an empty object for `lifecycle` in the container spec but this doesn't seem to cause any api errors.